integration-test/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitSpecificIntegrationTestBase.java

```diff
@@ -2479,6 +2481,71 @@ public void get_PutCalledBefore_ShouldGet() throws TransactionException {
     assertThat(getBalance(result.get())).isEqualTo(1);
   }
 
+  @Test
+  public void get_PutThenGetWithConjunctionReturnEmptyFromStorage_ShouldReturnEmpty()
+      throws TransactionException {
+    // Arrange
+    DistributedTransaction transaction = manager.begin();
+
+    // Act
+    transaction.put(preparePut(0, 0, namespace1, TABLE_1).withValue(BALANCE, 1));
+    Get get =
+        Get.newBuilder(prepareGet(0, 0, namespace1, TABLE_1))
+            .where(ConditionBuilder.column(BALANCE).isEqualToInt(1))
+            .build();
+    Optional<Result> result = transaction.get(get);
+    assertThatCode(transaction::commit).doesNotThrowAnyException();
+
+    // Assert
+    assertThat(result).isNotPresent();
+  }
+
+  @Test
+  public void
+      get_PutThenGetWithConjunctionReturnResultFromStorageAndMatchedWithPut_ShouldReturnResult()
+          throws TransactionException {
+    // Arrange
+    populateRecords(namespace1, TABLE_1);
+    DistributedTransaction transaction = manager.begin();
+    Put put = preparePut(0, 0, namespace1, TABLE_1).withValue(BALANCE, 1);
+    Get get =
+        Get.newBuilder(prepareGet(0, 0, namespace1, TABLE_1))
+            .where(ConditionBuilder.column(BALANCE).isLessThanOrEqualToInt(INITIAL_BALANCE))
+            .build();
+
+    // Act
+    transaction.put(put);
+    Optional<Result> result = transaction.get(get);
+    assertThatCode(transaction::commit).doesNotThrowAnyException();
+
+    // Assert
+    assertThat(result).isPresent();
+    assertThat(getBalance(result.get())).isEqualTo(1);
+  }
+
+  @Test
+  public void
+      get_PutThenGetWithConjunctionReturnResultFromStorageButUnmatchedWithPut_ShouldReturnEmpty()
```

Comment:
This specification might be somewhat unintentional for some users since the result is considered empty regardless of the result from storage if the result merged with the write set does not match the conjunction. However, I think it is a corner case and unavoidable.